--- src/gui/itemviews/qtreeview.cpp.orig	2008-07-29 12:26:54.000000000 +0200
+++ src/gui/itemviews/qtreeview.cpp	2008-10-19 14:04:14.234375000 +0200
@@ -1928,6 +1928,8 @@
 {
     Q_D(QTreeView);
     d->viewItems.clear(); // prepare for new layout
+    qSort(d->expandedIndexes.begin(), d->expandedIndexes.end());
+    qSort(d->hiddenIndexes.begin(), d->hiddenIndexes.end());
     QModelIndex parent = d->root;
     if (d->model->hasChildren(parent)) {
         d->layout(-1);
@@ -1935,8 +1937,6 @@
     }
     QAbstractItemView::doItemsLayout();
     d->header->doItemsLayout();
-    qSort(d->expandedIndexes.begin(), d->expandedIndexes.end());
-    qSort(d->hiddenIndexes.begin(), d->hiddenIndexes.end());
 }
 
 /*!
